### DISCUSSION TITLE
Add Object Declaration in G-Code Output

### DISCUSSION THREAD

**Discussion by erebus04** [OrcaSlicer/OrcaSlicer#76] [Ideas]
In SuperSlicer there is a function in the output options "Declare objects". 
This is very helpful with a printer that runs with Klipper and has a very large print bed. 
Here you can call a macro at the start sequence that creates the mesh only in the size of the required object and does not create the mesh for the entire print bed.

Description of the Macro https://github.com/kyleisah/Klipper-Adaptive-Meshing-Purging

Pictures from SuperSlicer and the G-Code that generated
![Declare-Slicer](https://user-images.githubusercontent.com/34960970/210049813-eaf537c9-26e9-4a8c-8c84-d6419d0c7443.png)

![Declare-G-Code](https://user-images.githubusercontent.com/34960970/210049815-3b21ce63-65ef-4e16-bc42-b2770c461436.png)

**Comment by SoftFever** [comment]
This has been supported.
`Label Object` is enabled by default.

To enable `Exclude Object` feature in Klipper: [follow this instruction](https://docs.mainsail.xyz/features/exclude_objects)

**Reply by erebus04** [reply]
Thank you,

I didn't see the button. Blind😬🙈

**Comment by TacticallyGrumpy** [comment]
Sorry to res this old thread but I figure it's better than making a new one. ` Label Object ` doesn't seem to be working for me in regards to ` Exclude Object `. It works correctly with SS. Is this still working in 1.4.5?

**Reply by TacticallyGrumpy** [reply]
It just started working now.... I didn't change anything other than filament. I have no idea what could have made the difference.
